content/react.md

```diff
@@ -0,0 +1,271 @@
+
+Meteor.startup(() => {
+  render(HelloWorld, document.getElementById('app'));
+});
+```
+
+XXX: should we be getting rid of `blaze-html-templates` and replacing with `static-html`? The tutorial doesn't sem to do it..
+
+<h3 id="using-third-party-npm-packages">Using 3rd party packages</h3>
+
+If you'd like to use a third party React component that has been published on NPM (such as the ones you find on the [React Components site](http://react-components.com)), you can simple `npm install --save` them and `import` from within your app.
+
+If you are looking to write a Atmosphere package that wraps such a component, you need to take some [further steps](#atmosphere-packages).
+
+<h3 id="using-with-blaze">Using Blaze with React</h3>
+
+If you'd like to use React within a larger app built with [Blaze](#blaze.html), you can use the [`react-template-helper`](https://atmospherejs.com/meteor/react-template-helper) component which renders a react component inside a Blaze template. First run `meteor add react-template-helper`, then user the `React` helper in your template:
+
+```html
+<template name="userDisplay">
+  <div>Hello, {{username}}</div>
+  <div>{{> React component=UserAvatar userId=_id}}</div>
+</template>
+```
+
+You will need to pass in the component class with a helper:
+
+```js
+import { Template } from 'meteor/templating';
+
+import 'userDisplay.html';
+import UserAvatar from './UserAvatar.jsx';
+
+Template.userDisplay.helpers({
+  UserAvatar() {
+    return UserAvatar;
+  }
+})
+```
+
+The `component` argument is the React component to include, which should be passed in with a helper.
+
+Every other argument is passed as a prop to the component when it is rendered.
+
+Note that there a few caveats:
+
+ - React components must be the only thing in the wrapper element, due to a limitation of React (see facebook/react [#1970](https://github.com/facebook/react/issues/1970), [#2484](https://github.com/facebook/react/issues/2484)), a React component must be rendered as the only child of its parent node, meaning it cannot have any siblings.
+
+ - This means a component also can't be the only thing in a template, because it's impossible to tell where the template will be used.
+
+<h4 id="passing-callbacks-from-blaze">Passing callbacks to a React component</h4>
+
+To pass a callback to a React component that you are including with this helper, simply make a [template helper that returns a function](http://guide.meteor.com/blaze.html#pass-callbacks), and pass it in as a prop, like so:
+
+```js
+Template.userDisplay.helpers({
+  onClick() {
+    var self = Template.instance();
+
+    // Return a function from this helper, where the template instance is in
+    // a closure
+    return function () {
+      self.hasBeenClicked.set(true)
+    }
+  }
+});
+```
+
+To use it in Blaze:
+
+```js
+<template name="userDisplay">
+  <div>{{> React component=UserAvatar userId=_id onClick=onClick}}</div>
+</template>
+```
+
+<h2 id="livedata">Using data</h2>
```

Comment:
This title is a bit vague, I think it should be improved to maybe "data loading" or "integration with Meteor data system" or something